spec/integration/food/food_index_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe 'Food', type: :system do
+  before(:each) do
+    RecipeFood.delete_all
+    Food.delete_all
+    Recipe.delete_all
+    User.delete_all
+    @user = User.create(name: 'User', surname: 'Spec', email: 'testing@gmail.com', password: 'testing')
+    @r1 = Recipe.create(user: @user, name: 'Recipe one', preparation_time: 1, cooking_time: 2,
+                        description: 'Recipe one description', public: true)
+    @f1 = Food.create(user: @user, name: 'Food item', measurement_unit: 'kg', price: 2, quantity: 5)
+  end
+
+  it 'Shows the correct index view' do
+    visit new_user_session_path
+    fill_in 'Email', with: @user.email
+    fill_in 'Password', with: @user.password
+    click_button 'Log in'
+    sleep(1)
```

Comment:
👍 Good progress on your project! 👍 However, there's a suggestion for improving the code:

- [ ] **DRY Up Login Code:** In multiple places, the same code for user login is repeated. According to the DRY (Don't Repeat Yourself) principle, it's a good practice to avoid redundancy and keep the codebase clean. You can create a helper method for user login to streamline the login process.

Here's an example of how you can create a helper method for logging in:

```ruby
# spec/support/login_helper.rb

module LoginHelper
  def login_user(user)
    visit new_user_session_path

    fill_in 'Email', with: user.email
    fill_in 'Password', with: 'testing'

    click_button 'Log in'
    sleep(1)
  end
end
```

Then, include this module in your `rails_helper.rb`:

```ruby
# spec/rails_helper.rb

RSpec.configure do |config|
  # ...

  config.include LoginHelper, type: :system
end
```

Now, in your tests, you can simply call the `login_user` method:

```ruby
context 'user logs in successfully' do
  it 'should redirect to the root path' do
    login_user(@user)
    expect(current_path).to eq(root_path)
    expect(page).to have_content('Signed in successfully.')
  end
end
```

This approach will make your code more maintainable and adherent to best practices. Keep up the good work! 🚀👩‍💻